### ISSUE TITLE
[🐛] Error processing request body: Could not find image file

### ISSUE THREAD

**Issue by dmsierra11** [react-native-image-picker/react-native-image-picker#2119]
### Description

Error when trying to upload image from gallery

![Simulator Screen Shot - iPhone 12 - 2023-04-20 at 13 10 32](https://user-images.githubusercontent.com/2533919/233573856-6e5f8e33-36b6-4e06-a1bf-8b5ffb2a14d0.png)

### How to repeat issue and example

```
const imagePickerDefaultOptions = {
  selectionLimit: 0,
  mediaType: 'photo' as MediaType,
  includeBase64: false,
  includeExtra: true,
};

launchImageLibrary(imagePickerDefaultOptions, (imgPickerResponse) => {
    if (imgPickerResponse.assets?.[0]) {
      onPhotoTaken(imgPickerResponse.assets?.[0]);
    }
  });

```
### Solution

What needs to be done to address this issue?

### Additional Information

- Image Picker version: 4.10.3
- React Native version: 0.66.1
- Platform: iOS
- Development Operating System: MacOS
- Dev tools: Xcode 14.0.1, iOS 16.0

**Comment by yorickshan** [comment]
same issue

**Comment by VanDinh18** [comment]
same issue

**Comment by cokron** [comment]
Same here - upgraded from react-native-image-picker 4.10.3 to  5.3.1. 
It used to work in 4.10.3 both in Simulator and on real device (iPhone).
Now it stopped working. Going back to 4.10.3 for now.

**Comment by jeremiahlachica** [comment]
I also have the same issue. Currently using 5.3.1

**Comment by maydersonmellops** [comment]
Same error here, until this correction was made, I used lib `react-native-image-crop-picker` and implemented it as follows:

```tsx
import ImageCropPicke from 'react-native-image-crop-picker';

ImageCropPicker.openPicker({
  mediaType,
   cropping: false,
   compressImageQuality: 0.8,
})
   .then(data => console.log(data))
   .catch(error => console.log(error));
```

Had a similar result to Image Picker.

**Comment by maydersonmellops** [comment]
@dmsierra11 I believe the problem is here.

Successfully selected image:

```json
{
  "fileName":"5693202E-E952-4C0D-A69B-2FF9B0F6C688.jpg",
  "fileSize":993045,
  "height":2500,
  "type":"image/jpg",
  "uri":"file:///Users/mayderson/Library/Developer/CoreSimulator/Devices/D555E641-C03B-407D-9251-28DBEF5BFEE6/data/Containers/Data/Application/D641C420-4AE4-49AB-A63B-5B05E88F59D7/tmp/5693202E-E952-4C0D-A69B-2FF9B0F6C688.jpg",
  "width":1668
}
```

Error selected image:

```json
{
  "fileName":"B22A70A5-D144-4857-8249-97C1FDB3FAE1.jpg",
  "height":0,
  "type":"image/jpg",
  "uri":"file:///Users/mayderson/Library/Developer/CoreSimulator/Devices/D555E641-C03B-407D-9251-28DBEF5BFEE6/data/Containers/Data/Application/D641C420-4AE4-49AB-A63B-5B05E88F59D7/tmp/B22A70A5-D144-4857-8249-97C1FDB3FAE1.jpg",
  "width":0
}
```

In other words, the `height` and `width` attributes are coming with `0` and the `fileSize` is not coming, which is causing this problem.

**Comment by dmsierra11** [comment]
Thank you @maydersonmellops, I also noticed the `0` values on `height` and `width`, it looks like there is an error selecting the image from the library, only instead of giving an error on that part of the process it just gives the image path with these 0 values. Do you have any idea how to fix it? I hope the maintainer of this repo can give us some light on how to fix it. Meanwhile I will also try your suggestion and change to `react-native-image-crop-picker`

**Comment by dmsierra11** [comment]
@cokron it happens to me on `4.10.3` too, unfortunately.

**Comment by dmsierra11** [comment]
@maydersonmellops it actually happens the same to me with `react-native-image-crop-picker`

**Comment by bcgilliom** [comment]
Seems to only happen for me on the sim flower image (the one picked in the video above), which is marked as `HDR` and has image type of `HEIC` when viewed in the photos app

**Comment by minh-dai** [comment]
+1

**Comment by dmsierra11** [comment]
Yes @bcgilliom, you are totally right, it happens to me only on this picture too. I just don’t know why. FYI I tried with a different HEIC image and it worked.

**Comment by blyszcz** [comment]
I had the same experience with a sunflower image, curious why only on this image

**Comment by alizaib1217** [comment]
> I had a similar encounter with a captivating sunflower image. If anyone else has experienced this issue with different images, I would greatly appreciate it if you could kindly share your observations.

I had the same issue I believe its crashing for the image which are not available like the one that are on cloud. Once the cloud image is downloaded it does not crash.

**Comment by andresmejia3** [comment]
Is there any solution to this yet?

**Comment by andresmejia3** [comment]
Or any workaround?

**Comment by Acetyld** [comment]
Any work around?

**Comment by vksgautam1986** [comment]
3.6.0 is working fine but its quite older version

**Comment by jakobpesch** [comment]
Also happens in 5.4.0.

**Comment by NikitaRadzkov** [comment]
Also happens in 5.4.2

**Comment by dflm25** [comment]
Hello same issue, Working with "react-native-image-picker": "^4.10.3",
<img width="424" alt="Screenshot 2023-08-25 at 6 10 44 PM" src="https://github.com/react-native-image-picker/react-native-image-picker/assets/1731117/fa5076f2-64a7-4583-ba75-64590219d0c4">

**Comment by andresmejia3** [comment]
> same issue on ` "react-native-image-picker": "^7.0.0",`

Use `react-native-image-crop-picker`

**Comment by DusanSpasojevic98** [comment]
> > I had a similar encounter with a captivating sunflower image. If anyone else has experienced this issue with different images, I would greatly appreciate it if you could kindly share your observations.
> 
> I had the same issue I believe its crashing for the image which are not available like the one that are on cloud. Once the cloud image is downloaded it does not crash.

@alizaib1217 Is there any way to don't show cloud images or to download them when picked, or any other solution?

**Comment by DusanSpasojevic98** [comment]
This is my config on version 7.0.1  : 
    let response = await launchImageLibrary({
      includeBase64: true,
      mediaType: 'photo',
      quality: 0.5,
    });

And it works

**Comment by pnadj** [comment]
Same issue in 7.0.1

**Comment by samcodefish** [comment]
I have [created a PR](url) that hopefully fixes this issue. Any testing / feedback would be helpful.

**Comment by JB712** [comment]
Looking at @samcodefish [PR](https://github.com/react-native-image-picker/react-native-image-picker/pull/2236) and from my understanding of the problem, it seems that .heic files cannot be handled like .jpg files. When selecting a .heic file, the copy that you can access from _launchImageLibrary_ response is a .jpg file (at least it is what I find in uri property) and that is why height, width, type and fileSize are wrong and you can't access it.

Unfortunately, your PR does not correct this behavior, I still got uri to a .jpg file that result in a bug (_"Could not find image ..."_)

I just want to add that to get to this conclusion, I only used the .heic image from the iPhone 12 mini - iOS 16.2 emulator, the one that display flowers. Maybe it is different with a real device or with different images
